api/src/main/java/pl/kolendateam/dadcard/items/weapons/entity/Weapons.java

```diff
@@ -0,0 +1,53 @@
+package pl.kolendateam.dadcard.items.weapons.entity;
+import java.util.ArrayList;
+
+import io.micrometer.common.lang.NonNull;
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import pl.kolendateam.dadcard.items.entity.Items;
+
+
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@DiscriminatorValue(value = "WEAPON")
+public class Weapons extends Items {
+
+    @Enumerated(EnumType.STRING)
+    WeaponNameEnum weaponName;
+
+    @NonNull
+    @Enumerated(EnumType.STRING)
+    WeaponNumericEnum damage;
+
+    @NonNull
+    @Enumerated(EnumType.STRING)
+    WeaponNumericEnum critical;
+
+    Integer range;
+
+    @NonNull
+    String type;
+
+    String specialAttacks;
+
+    public int findItemIndexinArrayById(ArrayList<Object> items,Weapons weapon) {
+
+        for(int i = 0; i < items.size(); i++){
+            if(weapon.equals(items.get(i))){
+                return i;
+            }
+        }
+
+        return 0;
```

Comment:
return 0 dla findindex to zly pomysl bo index tablicy moz byc 0. np. w  js zwraca sie -1 albo mozna wolac wyjatek

Reply:
tak, poprawiłem na return -1